### PR TITLE
tests/gnrc_netif: add test for ULA source selection

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Adds a test case for ULA source selection. For me it passes, but this test is an edge-case, as link-local addresses (`fe80::/64`) and ULAs (`fd00::/8`) are very close to matching so if something is messed up in the implementation this might fail.

### Issues/PRs references
Discussed in https://github.com/RIOT-OS/RIOT/pull/9522#issuecomment-411688679

### CODE REVIEW

**Comment by cladmi** [comment]
The test is working for `samr21-xpro`, `iotlab-m3`, `iotlab-a8-m3` and `arduino-zero`.
But I cannot really review the code.

File: tests/gnrc_netif/main.c [hunk 1]
```diff
@@ -450,6 +450,28 @@ static void test_ipv6_addr_best_src__other_subnet(void)
...
+{
+    static const ipv6_addr_t ula_src = { .u8 = NETIF0_IPV6_ULA };
+    static const ipv6_addr_t ula_dst = { .u8 = { ULA1, ULA2, ULA3, ULA4,
+                                                 ULA5, ULA6, ULA7, ULA8,
+                                                 0, 0, 0, 0, 0, 0, 0, 1 } };
+    ipv6_addr_t *out = NULL;
+    int idx;
+
+    test_ipv6_addr_add__success();
+    TEST_ASSERT(0 <= (idx = gnrc_netif_ipv6_addr_add_internal(netifs[0], &ula_src, 64U,
```
Comment (smlng) [inline tests/gnrc_netif/main.c]: It might be good to also add the case with address (flags) set TENTATIVE and verify its not selected? Same when a "real" (non-ULA) global address is present. Also with and w/o default route set.

Comment (miri64) [reply tests/gnrc_netif/main.c]: Sounds like a good idea, however this seems to be a different topic than this PR (which I would like to merge before doing another to avoid merge conflicts) ;-).

Comment (smlng) [reply tests/gnrc_netif/main.c]: well the TENTATIVE case should be tested for ULAs too, bc that's what we also want to verify, right? Adding tests for real/global addresses could be another PR.

**Comment by cladmi** [comment]
I would say, we merge it this way. Adding other tests if needed can be done in followup PRs.
